source/extensions/filters/http/jwt_authn/jwks.cc

```diff
@@ -0,0 +1,254 @@
+#include "extensions/filters/http/jwt_authn/jwks.h"
+
+#include "common/common/assert.h"
+#include "common/common/base64.h"
+#include "common/json/json_loader.h"
+
+#include "extensions/filters/http/jwt_authn/utils.h"
+
+#include "openssl/bn.h"
+#include "openssl/ecdsa.h"
+#include "openssl/evp.h"
+#include "openssl/rsa.h"
+#include "openssl/sha.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace JwtAuthn {
+
+namespace {
+
+/** Class to create EVP_PKEY object from string of public key, formatted in PEM or JWKs.
+ * If it failed, status_ holds the failure reason.
+ *
+ * Usage example:
+ * EvpPkeyGetter e;
+ * bssl::UniquePtr<EVP_PKEY> pkey = e.createEvpPkeyFromStr(pem_formatted_public_key);
+ * (You can use createEvpPkeyFromJwkRSA() or createEcKeyFromJwkEC() for JWKs)
```

Comment:
I'm not a deep expert in this area, but nothing in the above comment made sense to me :) I put it down to my own lack of knowledge here, but I think if you can explain at least what an EvpPkey is and what it's used for, that would be good for other casual readers of the Envoy sourc code.